cmd/zed/import.go

```diff
@@ -140,13 +140,38 @@ func importRelationships(client *authzed.Client, relationships string) error {
 	if err := scanner.Err(); err != nil {
 		return err
 	}
+	log.Info().Int("count", len(relationshipUpdates)).Msg("parsed relationships")
 
-	request := &v1.WriteRelationshipsRequest{Updates: relationshipUpdates}
-	log.Trace().Interface("request", request).Msg("writing relationships")
-	log.Info().Int("count", len(relationshipUpdates)).Msg("importing relationships")
+	const importBatchSize = 5000
+	return batchRequests(client, relationshipUpdates, importBatchSize)
+}
 
-	if _, err := client.WriteRelationships(context.Background(), request); err != nil {
-		return err
+func batchRequests(client *authzed.Client, updates []*v1.RelationshipUpdate, batchSize int) error {
+	// Number of batches required to import all relationships.
+	totalBatches := int(float64(len(updates))/float64(batchSize)+0.5) + 1
+	batchCount := 1
+
+	for i := 0; i < len(updates); i += batchSize {
+		end := i + batchSize
+		if end > len(updates) {
+			end = len(updates)
+		}
+
+		request := &v1.WriteRelationshipsRequest{Updates: updates[i:end]}
```

Comment:
You'd have a lot less to keep track of with something like:

```suggestion
        for i := 0; i < len(updates)/batchSize; i++ {
                request := &v1.WriteRelationshipsRequest{Updates: updates[i*batchSize : (i+1)*batchSize]}
```